libindy/src/services/metrics/mod.rs

```diff
@@ -12,32 +12,24 @@ pub mod models;
 const COMMANDS_COUNT: usize = MetricsService::commands_count();
 
 pub struct MetricsService {
-    queued_commands_count: RefCell<[u128; COMMANDS_COUNT]>,
-    queued_commands_duration_ms: RefCell<[u128; COMMANDS_COUNT]>,
-
-    executed_commands_count: RefCell<[u128; COMMANDS_COUNT]>,
-    executed_commands_duration_ms: RefCell<[u128; COMMANDS_COUNT]>,
+    queued_counters: RefCell<[CommandCounters; COMMANDS_COUNT]>,
+    executed_counters: RefCell<[CommandCounters; COMMANDS_COUNT]>,
 }
 
 impl MetricsService {
     pub fn new() -> Self {
         MetricsService {
-            queued_commands_count: RefCell::new([u128::MIN; COMMANDS_COUNT]),
-            queued_commands_duration_ms: RefCell::new([u128::MIN; COMMANDS_COUNT]),
-
-            executed_commands_count: RefCell::new([u128::MIN; COMMANDS_COUNT]),
-            executed_commands_duration_ms: RefCell::new([u128::MIN; COMMANDS_COUNT]),
+            queued_counters: RefCell::new([CommandCounters::new(0,0,[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]); COMMANDS_COUNT]),
+            executed_counters: RefCell::new([CommandCounters::new(0,0,[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]); COMMANDS_COUNT]),
```

Comment:
I suggest replacing `[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]` with `[0; BUCKET_COUNT]` for both counters.

Reply:
I think I have a better suggestion - remove parameters from `CommandCounters::new` method at all